tests/security/test_repo_setup.pm

```diff
@@ -0,0 +1,107 @@
+# Copyright © 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: Setup online repos or dvd images for further testing
+# Maintainer: wnereiz <wnereiz@member.fsf.org>
+# Tags: poo#52808
+
+use strict;
+use warnings;
+use base "consoletest";
+use testapi;
+use utils 'zypper_call';
+
+sub repo_cleanup {
+    assert_script_run "SUSEConnect --cleanup";
+    assert_script_run "rm -f /etc/zypp/repos.d/*.repo";
+    # Make sure all repositories are cleaned
+    validate_script_output("zypper lr || true", sub { m/No repo/ });
+}
+
+sub run {
+    my ($self) = @_;
+    select_console 'root-console';
+
+    # Note: Check the ticket in p.o.o for detailed descriptions of the
+    #       repository setup logic.
+
+    if (!get_var('SECTEST_DVD_SRC')) {
+        my $sc_out = script_output("SUSEConnect --status-text", proceed_on_failure => 1);
+
+        record_info "Not Registered", "System is not registered, add DVD image as repositories." if ($sc_out =~ /Not Registered/);
+
+        # If the the network is not available to check if it is registered.
+        # Then proceed following steps to add DVD repo.
+        record_info "Network Error", "SUSEConnect error: SocketError" if ($sc_out =~ /SocketError/);
+
+        if ($sc_out =~ /Status.*ACTIVE/) {
+            my $regcode = get_var("SCC_REGCODE");
+            ($regcode) = $sc_out =~ /Regcode: (.*)/ if (!$regcode);
+            die "SCC_REGCODE not specified" if (!$regcode);
+            repo_cleanup;
+            # Base system registration
+            assert_script_run("SUSEConnect -r $regcode", timeout => 180);
```

Comment:
I think here invoke function add_suseconnect_product will be better.

Reply:
`register_product` should be used here instead of `add_suse_connect_product`, the drawback is that it is not possible to set **timeout** by this way. Sometimes it is very slow to register. But anyway o.s.d is fast enough so I did the change.

Reply:
The verification run have also been updated